smoke-tests-otel-starter/spring-smoke-testing/src/main/java/io/opentelemetry/spring/smoketest/HttpSpanDataAssert.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.spring.smoketest;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.satisfies;
+
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.sdk.testing.assertj.ResourceAssert;
+import io.opentelemetry.sdk.testing.assertj.SpanDataAssert;
+import io.opentelemetry.semconv.HttpAttributes;
+import io.opentelemetry.semconv.UrlAttributes;
+import io.opentelemetry.semconv.incubating.HttpIncubatingAttributes;
+import java.util.function.Consumer;
+import org.assertj.core.api.AbstractLongAssert;
+
+public class HttpSpanDataAssert {
+
+  private final SpanDataAssert span;
+
+  private HttpSpanDataAssert(SpanDataAssert span) {
+    this.span = span;
+  }
+
+  public static HttpSpanDataAssert create(SpanDataAssert serverSpan) {
+    return new HttpSpanDataAssert(serverSpan);
+  }
+
+  @CanIgnoreReturnValue
+  public HttpSpanDataAssert assertClientGetRequest(String path) {
+    span.hasKind(SpanKind.CLIENT)
+        .hasAttributesSatisfying(
+            satisfies(UrlAttributes.URL_FULL, a -> a.endsWith(path)),
+            // this attribute is set by the experimental http instrumentation
+            satisfies(
+                HttpIncubatingAttributes.HTTP_RESPONSE_BODY_SIZE,
+                AbstractLongAssert::isNotNegative));
+    return this;
+  }
+
+  @CanIgnoreReturnValue
+  public HttpSpanDataAssert assertServerGetRequest(String route) {
+    span.hasKind(SpanKind.SERVER)
+        .hasAttribute(HttpAttributes.HTTP_REQUEST_METHOD, "GET")
+        .hasAttribute(HttpAttributes.HTTP_RESPONSE_STATUS_CODE, 200L)
+        .hasAttribute(HttpAttributes.HTTP_ROUTE, route)
```

Comment:
It is more common to add assertions inside `hasAttributesSatisfyingExactly` like in https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/47bb31dc99a3c1a4f0882021b981c7384159d3b3/instrumentation/jms/jms-1.1/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jms/v1_1/Jms1InstrumentationTest.java#L62 Although you are using `hasAttributesSatisfying` here where you don't need to add all assertions maybe it would still make sense to use the same approach here?

Reply:
tried it out, but it turns out there are may attributes that are not interesting here: 

java.lang.AssertionError: [span [GET /ping] attribute keys] 
Expecting actual:
  [user_agent.original,
    network.protocol.version,
    server.port,
    http.request.body.size,
    url.scheme,
    http.response.status_code,
    server.address,
    client.address,
    network.peer.address,
    http.response.body.size,
    url.path,
    http.request.method,
    http.route,
    network.peer.port]
to contain exactly in any order:
  [http.response.body.size,
    http.request.method,
    http.route,
    http.response.status_code]
but the following elements were unexpected:
  [user_agent.original,
    network.protocol.version,
    server.port,
    http.request.body.size,
    url.scheme,
    server.address,
    client.address,
    network.peer.address,
    url.path,
    network.peer.port]

Reply:
if you don't want to assert all the attributes then you should use `hasAttributesSatisfying` instead of `hasAttributesSatisfyingExactly`